client/state/selectors/test/can-upgrade-to-plan.js

```diff
@@ -0,0 +1,242 @@
+			)
+		);
+	} );
+
+	test( 'should return true from 1-year plans to 2-year plans', () => {
+		[
+			[ PLAN_PERSONAL, PLAN_PERSONAL_2_YEARS ],
+			[ PLAN_PREMIUM, PLAN_PREMIUM_2_YEARS ],
+			[ PLAN_BUSINESS, PLAN_BUSINESS_2_YEARS ],
+		].forEach( ( [ planOwned, planToPurchase ] ) =>
+			expect( canUpgradeToPlan( makeState( siteId, planOwned ), siteId, planToPurchase ) ).toBe(
+				true
+			)
+		);
+	} );
+
+	test( 'should return false from 2-year plans to 1-year plans', () => {
+		[
+			[ PLAN_PERSONAL_2_YEARS, PLAN_PERSONAL ],
+			[ PLAN_PREMIUM_2_YEARS, PLAN_PREMIUM ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_BUSINESS ],
+		].forEach( ( [ planOwned, planToPurchase ] ) =>
+			expect( canUpgradeToPlan( makeState( siteId, planOwned ), siteId, planToPurchase ) ).toBe(
+				false
+			)
+		);
+	} );
+
+	test( 'should return false from high-tier plans to lower-tier plans', () => {
+		[
+			[ PLAN_BUSINESS, PLAN_FREE ],
+			[ PLAN_BUSINESS, PLAN_PERSONAL ],
+			[ PLAN_BUSINESS, PLAN_PREMIUM ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_FREE ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_PERSONAL ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_PERSONAL_2_YEARS ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_PREMIUM ],
+			[ PLAN_BUSINESS_2_YEARS, PLAN_PREMIUM_2_YEARS ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_PERSONAL_MONTHLY ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_PREMIUM ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_PREMIUM_MONTHLY ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_PERSONAL_MONTHLY ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_PREMIUM ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_PREMIUM_MONTHLY ],
+			[ PLAN_JETPACK_PERSONAL, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_PERSONAL_MONTHLY, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_PREMIUM, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_PREMIUM, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_PREMIUM, PLAN_JETPACK_PERSONAL_MONTHLY ],
+			[ PLAN_JETPACK_PREMIUM_MONTHLY, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_PREMIUM_MONTHLY, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_PREMIUM_MONTHLY, PLAN_JETPACK_PERSONAL_MONTHLY ],
+			[ PLAN_PERSONAL, PLAN_FREE ],
+			[ PLAN_PERSONAL_2_YEARS, PLAN_FREE ],
+			[ PLAN_PREMIUM, PLAN_FREE ],
+			[ PLAN_PREMIUM, PLAN_PERSONAL ],
+			[ PLAN_PREMIUM_2_YEARS, PLAN_FREE ],
+			[ PLAN_PREMIUM_2_YEARS, PLAN_PERSONAL ],
+			[ PLAN_PREMIUM_2_YEARS, PLAN_PERSONAL_2_YEARS ],
+		].forEach( ( [ planOwned, planToPurchase ] ) =>
+			expect( canUpgradeToPlan( makeState( siteId, planOwned ), siteId, planToPurchase ) ).toBe(
+				false
+			)
+		);
+	} );
+
+	test( 'should return true from high-tier expired plans to lower-tier plans', () => {
+		const makeComplexState = ( s, productSlug, isJetpack, isAtomic ) => ( {
+			sites: {
+				items: {
+					[ s ]: {
+						jetpack: isJetpack || isAtomic,
```

Comment:
This flirts with too-magical for unit test (I know, I introduced it 🙄 )

But, `atomic` implies `Jetpack`, so it works. Might make sense to represent that more clearly with something like `siteType = [ 'simple' (default) | 'jetpack' | 'atomic' ]`